src/multitask_personalization/envs/pybullet/pybullet_csp.py

```diff
@@ -128,36 +127,24 @@ def _get_place_plan(self, obs: PyBulletState) -> list[PyBulletAction]:
             surface_extents[1] / 2 - object_extents[1] / 2,
             surface_extents[2] / 2 + object_extents[2] / 2,
         )
-        surface_pose = get_pose(surface_id, self._sim.physics_client_id)
-        collision_ids = self._sim.get_collision_ids() - {surface_id, held_obj_id}
-        selected_placement: Pose | None = None
         for _ in range(100000):
             placement_position = self._rng.uniform(placement_lb, placement_ub)
             relative_placement_pose = Pose.from_rpy(
                 tuple(placement_position), (0, 0, np.pi / 2)
             )
-            placement_pose = multiply_poses(surface_pose, relative_placement_pose)
-            set_pose(held_obj_id, placement_pose, self._sim.physics_client_id)
-            has_collision = False
-            for collision_id in collision_ids:
-                if check_body_collisions(
-                    collision_id, held_obj_id, self._sim.physics_client_id
-                ):
-                    has_collision = True
-                    break
-            if not has_collision:
-                selected_placement = relative_placement_pose
-                break
-        assert selected_placement is not None
-        self._sim.set_state(obs)
-        return get_plan_to_place_object(
-            obs,
-            held_obj,
-            surface_name,
-            selected_placement,
-            self._sim,
-            max_motion_planning_candidates=self._max_motion_planning_candidates,
-        )
+            self._sim.set_state(obs)
+            plan = get_plan_to_place_object(
```

Comment:
this change was the result of some painful debugging. note to future self: it is possible that an IK solution exists for a placement pose, but not for a pre-placement pose, even if the pre-placement pose is very close to the placement pose...